packages/hardhat-core/test/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -365,6 +369,374 @@ describe("Hardhat module", function () {
           });
         }
       });
+
+      describe("hardhat_setBalance", function () {
+        it("should reject an invalid address", async function () {
+          await assertInvalidArgumentsError(
+            this.provider,
+            "hardhat_setBalance",
+            ["0x1234", "0x0"],
+            'Errors encountered in param 0: Invalid value "0x1234" supplied to : ADDRESS'
+          );
+        });
+
+        it("should reject a non-numeric nonce", async function () {
+          await assertInvalidArgumentsError(
+            this.provider,
+            "hardhat_setBalance",
+            [DEFAULT_ACCOUNTS_ADDRESSES[0].toString(), "xyz"],
+            'Errors encountered in param 1: Invalid value "xyz" supplied to : QUANTITY'
+          );
+        });
+
+        it("should not reject valid argument types", async function () {
+          await this.provider.send("hardhat_setBalance", [
+            DEFAULT_ACCOUNTS_ADDRESSES[0].toString(),
+            "0x0",
+          ]);
+        });
+
+        it("should result in a modified balance", async function () {
+          // Arrange: Capture existing balance
+          const existingBalance = new BN(
+            (
+              await this.provider.send("eth_getBalance", [
+                DEFAULT_ACCOUNTS_ADDRESSES[0],
+              ])
+            ).replace("0x", ""),
+            "hex"
+          );
+
+          // Act: Set the new balance.
+          const targetBalance = existingBalance.add(new BN(1)).mul(new BN(2));
+          // For sanity, ensure that we really are making a change:
+          assert.notEqual(targetBalance, existingBalance);
+          await this.provider.send("hardhat_setBalance", [
+            DEFAULT_ACCOUNTS_ADDRESSES[0],
+            `0x${targetBalance.toString(16)}`,
+          ]);
+
+          // Assert: Ensure the new balance was set.
+          const newBalance = new BN(
+            (
+              await this.provider.send("eth_getBalance", [
+                DEFAULT_ACCOUNTS_ADDRESSES[0],
+              ])
+            ).replace("0x", ""),
+            "hex"
+          );
+          assert(targetBalance.eq(newBalance));
+        });
+
+        it("should not result in a modified state root", async function () {
+          // Arrange 1: Send a transaction, in order to ensure a pre-existing
+          // state root.
+          await this.provider.send("eth_sendTransaction", [
+            {
+              from: DEFAULT_ACCOUNTS_ADDRESSES[0],
+              to: DEFAULT_ACCOUNTS_ADDRESSES[1],
+              value: "0x100",
+            },
+          ]);
+          await this.provider.send("evm_mine");
```

Comment:
Is this necessary here? Or is automine enabled?

Reply:
Yeah, automine is the default. No need to do this.

Reply:
Resolved via https://github.com/nomiclabs/hardhat/pull/1435/commits/5ef98460a0752336f3c134e48e958411f0144c62